### ISSUE TITLE
Orion-LD breaks on deployment

### ISSUE THREAD

**Issue by dalxds** [Fiware/context.Orion-LD#925]
I try to deploy Orion-LD with the helm chart using the latest image, troe enabled.
The pod running the Orion-LD crashed and the logs show the following:

```
orionld: value 7102819 for option '-ctxTimeout' not within limits
         0 <= 7102819 <= 20000 (default value)

Usage: orionld  [option '-U' (extended usage)]
                [option '-u' (usage)]
                [option '--version' (show version)]
                [option '-logDir' <log file directory>]
                [option '-t' <trace level>]
                [option '-logLevel' <initial log level (NONE, FATAL, ERROR, WARN, INFO, DEBUG)>]
                [option '-logAppend' (append to log-file)]
...
```

The chart doesn't provide a field for `ctxTimeout` and I couldn't locate it on the code. 
I don't know how it is set. 

Any help would be appreciated. 
Thanks!

**Comment by kzangeli** [comment]
That is beyond weird ...
The variable `contextDownloadTimeout`, connected to the CLI option `-ctxTimeout`, is found in `src/app/orionld/orionld.cpp`, and it has a default value of 5000 (milliseconds) and the limits are 0 and 20000 (milliseconds).

It would be quite nice to understand how you're getting this error !
As a workaround, try setting it to 2000 (-ctxTimeout 2000), and see what happens.
Probably something else crazy will happen instead - this all smells of some kind of memory corruption.
How much RAM does the broker have ?

**Comment by dalxds** [comment]
Hello @kzangeli.
Thanks for looking into it!

To help reproduce the error, I'm using the following `Chart.yaml` and `values.yaml`:
```yaml
apiVersion: v2
name: neptune-eew
description: A Helm Chart to deploy neptune-eew.
type: application
version: 0.1.0
appVersion: "0.1.0"
dependencies:
- name: orion
  version: "*"
  repository: https://fiware.github.io/helm-charts
  tags: 
    - fiware-core
- name: mongodb
  version: "*"
  repository: https://charts.bitnami.com/bitnami
  tags: 
    - fiware-core
- name: timescaledb-single
  version: "*"
  repository: https://charts.timescale.com
  tags:
    - fiware-core
```
``` yaml
mongodb:
  architecture: standalone
  useStatefulSet: true
  auth:
    enabled: false
  tls:
    enabled: false
    autoGenerated: false
  podSecurityContext:
    enabled: false
  containerSecurityContext:
    enabled: false

orion:
  broker:
    troe:
      enabled: true
      dbHost: neptune-eew-timescaledb-0.neptune-eew-replica.default.svc.cluster.local
      dbUser: postgres
      dbPort: 5432
      dbPassword: gmruukctFW1j93e5r7RI5ZIwLNvG5oxE
    mutexPolicy: none
    reqPoolSize: 32
    db:
      hosts:
        - neptune-eew-mongodb-0.neptune-eew-mongodb.default.svc.cluster.local
    logging:
      level: DEBUG

timescaledb-single:
  replicaCount: 1
  image:
    pullPolicy: IfNotPresent
  backup:
    enabled: false
  podManagementPolicy: Parallel
  serviceAccount:
    create: false
```

maybe I should set minimum resources for the deployment !?

**Comment by wistefan** [comment]
Hello @dalxds ,
if you don't set any resource limits, it will take all it gets. Thus, resources should not be the problem. I've never seen the parameter that you mention, thus its not part of the chart yet. 
Can you try to use a fixed version of orion-ld (e.g. 0.8.0 as its the last official release)? Be aware that "latest" is not guaranteed to be stable nor working at all, since its a preview version, pushed from development branches. 
I will take a look if I can reproduce the problem my self.

**Comment by dalxds** [comment]
Hello @wistefan,
thanks for the help.

Tried the `fiware/orion-ld:0.8.0` (latest stable beta), but I get the exact same error (with the same number as well).

**Comment by kzangeli** [comment]
I might have found the problem.
The size of the char buffer that stores the troe::dbHost is 64 bytes.
This value: `neptune-eew-mongodb-0.neptune-eew-mongodb.default.svc.cluster.local` is more than 64 bytes.

So, this might provoke a memory corruption in your system.
Easy fix, I'll have it pushed before EOB today.

**Comment by kzangeli** [comment]
Should be fixed by PR #942 .
Please test it and close the issue if all is OK

**Comment by kzangeli** [comment]
No news since Sep 1 2021 - assuming all is good.
Closing the issue
